node/core/pvf/src/worker_common.rs

```diff
@@ -394,3 +394,20 @@ pub async fn framed_recv(r: &mut (impl AsyncRead + Unpin)) -> io::Result<Vec<u8>
 	r.read_exact(&mut buf).await?;
 	Ok(buf)
 }
+
+// In case of node and worker version mismatch (as a result of in-place upgrade), send `SIGKILL`
+// to the node to tear it down and prevent it from raising disputes on valid candidates. Node
+// restart should be handled by the node owner. As node exits, unix sockets opened to workers
+// get closed by the OS and other workers receive error on socket read and also exit. Preparation
+// jobs are written to the temporary files that are renamed to real artifacts on the node side, so
+// no leftover artifacts are possible.
+pub(crate) fn kill_node_in_emergency() {
```

Comment:
Awesome, great comment! Some non-blockers:

nit 1: Docstrings should start with `///` not `//`.
nit 2: I would call this `kill_parent_node_in_emergency` to make it clear that this only works if called from a child worker.

Reply:
> Docstrings should start with `///` not `//`

But it's `pub(crate)`, does it make any sense to generate docs for it?

Reply:
I often do `cargo doc --open --document-private-items` to help me read through a module. Maybe a less-niche usecase is viewing the function doc in an IDE. Also, a private function might one day become public... I just default to always using `///`, don't see why not.